### PR TITLE
Blog app - Views

### PR DESCRIPTION
**Pull request by r-ahmed2022**
In this PR, I completed the following:-
- Implemented the design from the wireframes.
- image
- 
image

image

- Use methods that you have created in the Blog App 
-  Populate the database with dummy data

### CODE REVIEW

**Review by Whoistolu — COMMENTED** [review]
Hi @r-ahmed2022 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### You have submitted a project, that has an empty Live Demo section in the README. Kindly remove this section if this project has not been deployed in order to have a more professional README.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
+<div class="container mt-3">
+<% @users.each do |user| %>
+<div class="card text-center my-3">
+  <div class="card-header">
+  <%= user.id %>
+  </div>
+  <div class="card-body d-flex justify-content-start align-items-start">
+            <div class="user-photo">
+                <%= user.photo %>
+            </div>
```
Comment (TadesseDev) [inline app/views/users/index.html.erb:11]: - [ ] Dear @r-ahmed2022, Good job adding the `<%= user.photo %>` but doing it this way will only display the address for the image. Please try to render the photo in the `img` tag, that way the image will be visible.

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
+                <%= @user.photo %>
+            </div>
+            <div class="user_name-div">
+                <h1 class="user_name"> <%= @user.name %></h1>
+           </div>
+  </div>
+  <div class="card-footer text-muted d-flex justify-content-end">
+    <b>Number of posts:- </b> &nbsp; <%= @user.postscounter %>
+  </div>
+</div>
```
Comment (TadesseDev) [inline app/views/users/show.html.erb:17]: - [ ] Dear @r-ahmed2022, here you have rendered the number of posts, this is good as it is needed according to the wireframe. but we need to use `#recent_post` method to render the user's 3 most recent posts. Please use the  `#recent_post` and render the 3 most recent posts. 
### Your design 
image
### wireframe 
image

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1,17 +1,68 @@
...
+           
+            <div class="user_name-div">
+              
+             <%= post.text %>
+           </div>
+           <%= link_to 'See more..', post_path(post) %>
+  </div>
+  <div class="card-footer text-muted d-flex justify-content-end">
+    <b>Comments:- </b><span class="text-success"> &nbsp; <u>  <%= post.commentscounter %></u></span>&nbsp;&nbsp;
+    <b>Likes:- </b> &nbsp;<span class=" no-likes text-danger"><u> <%= post.likescounter %></u></span>
```
Comment (TadesseDev) [inline app/views/posts/index.html.erb:57]: - [ ] Dear r-ahmed2022, Here we are required to use the `post.recent_comments` to render 5 most recent comments as shown in the wireframe.

image

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @r-ahmed2022 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x] Good job overall following the template design ✅
 - [x]  Correct GitFlow  ✅
 - [x]  Good PR title/description  ✅
 - [x]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/index.html.erb (hunk 1 above)
Comment (r-ahmed2022) [reply app/views/users/index.html.erb:11]: @TadesseDev ,  in my opinion, we don't need to add photo at this PR, it is only placeholder , we need to focus view

File: app/views/posts/index.html.erb (hunk 3 above)
Comment (r-ahmed2022) [reply app/views/posts/index.html.erb:57]: @TadesseDev 
image

File: app/views/users/show.html.erb (hunk 2 above)
Comment (TadesseDev) [reply app/views/users/show.html.erb:17]: - [ ] Dear @r-ahmed2022, you are referring to the wireframe for the users/index page in your first image and the user/show in your second wireframe. These two need to be different. 
**here is the requirement I'm referring to, Please try your best to address it.**
image
image

File: app/views/posts/index.html.erb (hunk 3 above)
Comment (TadesseDev) [reply app/views/posts/index.html.erb:57]: - [ ] Dear @r-ahmed2022, also here, we are referring to the post/index page. and according to the requirement we need to use `post.recent_comments` to render the 5 most recent comments.

image

### Feel free to refer to the sample code.
image

> Please also remember we are here to help one another, And I'm only trying to make sure you have understood and completed the tasks as requested.

File: app/views/users/index.html.erb (hunk 1 above)
Comment (r-ahmed2022) [reply app/views/users/index.html.erb]: Done, as advised
image

File: app/views/users/show.html.erb (hunk 2 above)
Comment (r-ahmed2022) [reply app/views/users/show.html.erb:19]: Done, as advised
image

File: app/views/posts/index.html.erb (hunk 3 above)
Comment (r-ahmed2022) [reply app/views/posts/index.html.erb]: Done, as advised
image

**Review by DammyShittu — APPROVED** [review]
Hi @r-ahmed2022,

## APPROVED 🟢

## Highlights
✔️ Wireframe layout

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**